testing_support/mocks/api.go

```diff
@@ -33,8 +34,8 @@ type FakeMetricMetadataAPI struct {
 
 var _ api.MetricMetadataAPI = (*FakeMetricMetadataAPI)(nil)
 
-func NewFakeMetricMetadataAPI() FakeMetricMetadataAPI {
-	return FakeMetricMetadataAPI{
+func NewFakeMetricMetadataAPI() *FakeMetricMetadataAPI {
```

Comment:
Returning a pointer is more useful than the value, since the pointer implements the `MetadataAPI` interface. Everywhere that called this method had to use `&myAPI` everywhere, which was unnecessary.